File: Problems & Solutions/1833. Maximum Ice Cream Bars/06 January 2022/Approach 1 - Sorting (Greedy)/Maximum Ice Cream Bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        n, icecream = len(costs), 0

        while icecream < n and costs[icecream] <= coins:
            coins -= costs[icecream]
            icecream += 1

        return icecream

File: Problems & Solutions/1833. Maximum Ice Cream Bars/06 January 2022/Approach 1 - Sorting (Greedy)/Maximum Ice Cream Bars.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
        Arrays.sort(costs);

        int n = costs.length;
        int icecream = 0;

        while (icecream < n && costs[icecream] <= coins) {
            coins -= costs[icecream];
            icecream += 1;
        }

        return icecream;
    }
}